### ISSUE TITLE
Slickgrid with new Aurelia built in CLI bundler

### ISSUE THREAD

**Issue by zewa666** [ghiscoding/aurelia-slickgrid-demos#2]
Hey there,

I was trying to come up with a sample project built using the latest v2.0.0 CLI bundler and can't really figure out how to setup the dependencies.

I tried following the [descriptions from here](https://github.com/ghiscoding/aurelia-slickgrid/wiki/HOWTO--Step-by-Step) but didn't work out. So I tried not adding any dependencies and retry which resulted in the following:

<details><summary>Console errors</summary>

```
index.js:6 Uncaught ReferenceError: process is not defined
    at Object.<anonymous> (index.js:6)
    at Object.execCb (require.js:1696)
    at Module.check (require.js:883)
    at Module.enable (require.js:1176)
    at Object.enable (require.js:1557)
    at Module.<anonymous> (require.js:1161)
    at require.js:134
    at each (require.js:59)
    at Module.enable (require.js:1113)
    at Object.enable (require.js:1557)
(anonymous) @ index.js:6
... yadda yadda
Show 66 more frames


buffer_list.js:19 Uncaught (in promise) TypeError: Cannot read property 'inspect' of undefined
    at Object.<anonymous> (buffer_list.js:19)
    at Object.execCb (require.js:1696)
    at Module.check (require.js:878)
    at Module.enable (require.js:1176)
    at Object.enable (require.js:1557)
    at Module.<anonymous> (require.js:1161)
    at require.js:134
    at each (require.js:59)
    at Module.enable (require.js:1113)
    at Object.enable (require.js:1557)
(anonymous) @ buffer_list.js:19
... yadda yadda
Show 71 more frames
```
</details>

I've managed to overcome the process issue by following [this info](http://aurelia.io/docs/cli/cli-bundler/dependency-management#auto-tracing) to shim process using

```
import process from 'process';
window.process = process;
import {Buffer} from 'buffer';
window.Buffer = Buffer;
```

<details><summary>Console errors</summary>

```
Uncaught RangeError: Maximum call stack size exceeded
    at each (require.js:56)
    at Module.emit (require.js:1188)
    at Module.<anonymous> (require.js:1149)
    at require.js:134
    at require.js:1189
    at each (require.js:59)
    at Module.emit (require.js:1188)
    at Module.<anonymous> (require.js:1149)
    at require.js:134
    at require.js:1189
    ... yadda yadda ...

Uncaught (in promise) TypeError: Cannot read property 'inspect' of undefined
    at Object.<anonymous> (buffer_list.js:19)
    at Object.execCb (require.js:1696)
    at Module.check (require.js:878)
    at Module.enable (require.js:1176)
    at Object.enable (require.js:1557)
    at Module.<anonymous> (require.js:1161)
    at require.js:134
    at each (require.js:59)
    at Module.enable (require.js:1113)
    at Object.enable (require.js:1557)
    ... yadda yadda ...
```
</details>

So perhaps may I ask you to take a quick look by creating a fresh new CLI bundler TypeScript project and seeing whether you can make it run? Thanks in advance @ghiscoding

**Comment by ghiscoding** [comment]
Hmm I haven't tried at all Aurelia 2.0, are you trying with WebPack or something else? I think WebPack would be the easiest to test with.

**Comment by zewa666** [comment]
Hey, no I've meant with Aurelia v1 but the latest CLI version (2.0.0)

**Comment by ghiscoding** [comment]
oh I thought that was Au2, last time I did was around last December with CLI 1.4 I think. Can you try just to clone the repo and try it with that CLI version? It should run... I know 1 thing I didn't do yet is to upgrade to latest Au-I18N, I'm still on the 2.x version for that. I didn't know there was a 2.0 CLI

**Comment by zewa666** [comment]
yeah so the latest version of the CLI bundler (using makes under the hood) is going a bit different about the things where you typically don't need to touch too much, but it seems there is some trouble with the aurelia-slickgrid.

so I guess the easiest for you to try would be to run:

```
npx makes aurelia/v1 slickgrid-demo -s cli-bundler,typescript,scaffold-navigation
```

which will create a skeleton-navigation demo project and there you can add your dependencies as you see fit.

I'll try to ask @3cp, who created the new bundler, as well for help so maybe we can together figure it out.

**Comment by ghiscoding** [comment]
hmm ok I'm not sure when I'll have time to look at that, Aurelia is not for work and I'm busy with other things. However I wouldn't be surprised that it can't detect the necessary imports which I had to find them 1 by 1 and are displayed in the [HOWTO Step by Step - Wiki](https://github.com/ghiscoding/aurelia-slickgrid/wiki/HOWTO--Step-by-Step#2-for-aurelia-cli-with-requirejssystemjs-modify-the-aureliajson-file), have you tried to put them by hand in the `aurelia.json` file? You didn't mention WebPack anywhere, so I assume it's not? I only use WebPack on my side and I always find that I have to spend so much time to get the other bundlers working (SlickGrid is an old lib, it also uses jQuery, jQueryUI, so there's a lot of factors making it difficult to auto-detect stuff in bundlers)

**Comment by zewa666** [comment]
well the issue primarily comes up if original sources are webpack based because it starts mixing backend with frontend dependencies.

Yes I've tried adding them in one by one but it seems the issue is more specific to some backend requirements and/or endless loops.

**Comment by ghiscoding** [comment]
Ok that sux, infinite loop is a killer (literally lol), I'll have to try eventually but I'm not sure when I'll have time though. Do you have a link for the breaking changes in the new CLI? It's a bit annoying, that used to work so good before 😢

**Comment by 3cp** [comment]
The cli-bundler was not changed in latest cli v2.
It is a bug triggered in cli-bundler when tracing readable-stream. It's either overlooked or not hitting the problematic code path before.

https://github.com/nodejs/readable-stream/blob/7eded7bc298b14d1960e1db9ee48450bc61085ee/package.json#L55

By spec, it should return empty object, but I returned an undefined. https://github.com/defunctzombie/package-browser-field-spec/blob/master/README.md

Fix soon.

**Comment by 3cp** [comment]
Sorry, it seems I intend to return empty object. Something went wrong for the buffer_list.js file.

**Comment by 3cp** [comment]
Ok, the empty object/undefined is fixed in cli, will release soon. But I have no idea where the `Maximum call stack size exceeded.` came from (maybe some npm dep introduced circular dependency).

On the other hand, the demo is working with [dumber](https://github.com/dumberjs/new) bundler (the cli-bundler successor).

**Comment by 3cp** [comment]
FYI, this is not a new issue with aurelia-cli v2. The requirejs demo in this repo uses aurelia-cli v1, it fails the same thing.
The "makes" thing has nothing to do with cli-bundler, "makes" only provides new way to scaffold project.

**Comment by 3cp** [comment]
Yes, the circular dependency is between `excel-builder-webpacker/src/Excel/Drawing/index.js` and `excel-builder-webpacker/src/Excel/Drawing/Picture.js`.

My bet is this is broken since excel-builder-webpacker was introduced to this aurelia-slickgrid.

cli-bundler cannot handle this type of circular dependency. This is nothing we can do here, because the limitation is in `requirejs` (and also `alameda` and `systemjs`).

`dumber` bundler uses a different AMD module loader I wrote, it supports the CommonJS circular dependency.

**Comment by ghiscoding** [comment]
@3cp thanks for all the troubleshooting, but isn't there a way to manually include it like what I used to do it in CLI `1.x`? I used to do it as it's shown in my Wiki.. [HOWTO Step by Step (for RequireJS/SystemJS) - Wiki](https://github.com/ghiscoding/aurelia-slickgrid/wiki/HOWTO--Step-by-Step#2-for-aurelia-cli-with-requirejssystemjs-modify-the-aureliajson-file), is that now broken in `2.x`? If so that is kind of a regression... That was mainly to help others using my lib with RequireJS/SystemJS, but if I understand what you said then that means I would have to tell everyone that "hey guys only WebPack now works in CLI `2.x`"

I actually forked `excel-builder-webpack` and renamed it to `excel-builder-webpacker` but I don't want to put work in this lib, it was created long time ago and I only forked it to update some of the dependencies, I don't want to touch the code. This lib adds Excel Export to my lib (which most people prefer in a grid).

**Comment by 3cp** [comment]
You already updated your source code to use `excel-builder-webpacker`, not `excel-builder-webpack`.
The aurelia.json example showing `excel-builder-webpack` which is totally not used by any code.

Anyway, no aurelia.json config can bypass limitation of requirejs itself.

If you want to fix the circular deps without touching source code, you can use [@zeit/ncc](https://github.com/vercel/ncc) to pack the whole src into one `dist/index.js` file.

You may want to skip inner deps. It by default writes to `dist/index.js`.
```
ncc build src/index.js -m -e lodash -e jszip -e q
```

Or simply pack everything, and you need to move lodash,jszip,q from dependencies to devDependencies.
```
ncc build src/index.js -m
```

**Comment by 3cp** [comment]
You already has compiled code for excel-builder-webpacker in that npm package, just need to update your excel-builder-webpacker  package.json to point `main` to `dist/excel-builder.compiled.min.js`.

**Comment by 3cp** [comment]
Unfortunately, something is wrong with your compiled file.

```
Uncaught TypeError: o.extend is not a function
    at Object.require.100.lodash (excel-builder.compiled.min.js:1)
```

**Comment by 3cp** [comment]
I could not find a way to make it work with cli-bundler (even with ncc compiled), it must be hitting some other limitation of cli-bundler.

@zewa666 use dumber pls.

**Comment by ghiscoding** [comment]
The error says that Lodash is not a function, it is a dependency of the Excel-Builder but I'm not sure if the `o.extend` is actually jQuery though, my lib (Aurelia-Slickgrid) requires jQuery but if users wish to use the Excel Export, then that requires Lodash as well. I wish that both external libs would have used the same jQuery or Lodash, but they're not 😢

**Comment by 3cp** [comment]
As far as I can see, that `dist/excel-builder.compiled.min.js` is prepared by browserify which assume browser env without any module loader. It doesn't work in AMD environment. It intends to be loaded directly like in a `<script>` tag.

**Comment by 3cp** [comment]
I got a working ncc build for your excel-builder-webpacker.

**Comment by zewa666** [comment]
Wow thanks for the detailed explanation @3cp. I was under the wrong impression that makes automatically implied dumber, so thx for the clarification here. Anyways I remember the issues we had with Browserify packages from back then with some VSCode-Dependencies.

So did I get it right that after the referenced PR for excel-builder is merged we'd merely have to prepend it and keep the rest autodetected?

Thanks again for the hard work getting to the root-cause

**Comment by 3cp** [comment]
It should. Plus next aurelia-cli patch release, Rob is going to schedule that.

**Comment by 3cp** [comment]
@zewa666 nothing to be prepended.

**Comment by zewa666** [comment]
hmmm I wonder what needs to be done on aurelia-slickgrid side in order to make this work. I've tried cloning the slickgrid repo and pointing it to your fork. After a build it seems it still picks up something different because in my example I end up with

```
Uncaught RangeError: Maximum call stack size exceeded
and
Uncaught (in promise) TypeError: Drawing is not a constructor
```

which indicates its still going into the src and trying to directly compile it with Webpack. So I guess some webpack config is needed in order to tell it to pickup the resulting build from excel-exporter.js

**Comment by 3cp** [comment]
@zewa666 because the "main" was using src, not compiled file.
https://github.com/ghiscoding/excel-builder.js/pull/1/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2

**Comment by ghiscoding** [comment]
Wow that is amazing work there @3cp, I rarely get contributions in my lib so this is really really great to see all your efforts being put in finding and fixing the issue. I'll take couple of days to test this with all the bundlers and also all the libs that I have using this Excel Builder you updated... thanks again, this is amazing. 

I'll let you guys know soon, it should be end of week

**Comment by zewa666** [comment]
@3cp thats the fun part, I did install your fork instead of the original npm package for my test. But yeah, lets see how it turns out once it gets merged. Perhaps my issue was related to something else.

**Comment by ghiscoding** [comment]
@3cp I tried the compiled file directly into my Aurelia-Slickgrid lib and it doesn't run I get a few WebPack errors as the following
```shell
bluebird.js:1590 Unhandled rejection TypeError: __webpack_modules__[moduleId] is not a function    
at __webpack_require__ (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:75130:42)    
at Object.804 (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:75098:18)    
at __webpack_require__ (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:75130:42)    
at Object.861 (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:72450:9)    
at __webpack_require__ (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:75130:42)    
at Object.351 (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:75058:15)   
 at __webpack_require__ (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:75130:42)    
at http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:75147:18    
at Object.<anonymous> (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:75148:12)    
at Object.393 (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:75150:30)    
at __webpack_require__ (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:20:30)    
at Object.<anonymous> (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:134675:82)    
at Object.51 (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:135255:30)    
at __webpack_require__ (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:20:30)   
at Object.23 (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:54613:79)    
at __webpack_require__ (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:20:30)    
at Object.<anonymous> (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:154954:75)    
at Object.aurelia-slickgrid/custom-elements/aurelia-slickgrid (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:156010:30)    
at __webpack_require__ (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:20:30)    
at Object.aurelia-slickgrid/index (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:156248:93)    
at __webpack_require__ (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:20:30)    
at WebpackLoader.<anonymous> (http://localhost:9000/app.2fe5b8d9ab2fcb55846a.bundle.js:138769:51)
```
and if I click on the first link, it seems to be coming from the `Q` library that is a dependency of Excel-Builder
```js
module.exports = __webpack_require__(/*! q */ 400);

/***/ })

/******/ 	});
/************************************************************************/
/******/ 	// The module cache
/******/ 	var __webpack_module_cache__ = {};
/******/ 	
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/ 		// Check if module is in cache
/******/ 		if(__webpack_module_cache__[moduleId]) {
/******/ 			return __webpack_module_cache__[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = __webpack_module_cache__[moduleId] = {
/******/ 			// no module.id needed
/******/ 			// no module.loaded needed
/******/ 			exports: {}
/******/ 		};
/******/ 	
/******/ 		// Execute the module function
/******/ 		var threw = true;
/******/ 		try {
                             // THE LINE BELOW IS WHERE THE ERROR IS TROWN [moduleId]
/******/ 			__webpack_modules__[moduleId](module, module.exports, __webpack_require__);
/******/ 			threw = false;
/******/ 		} finally {
/******/ 			if(threw) delete __webpack_module_cache__[moduleId];
/******/ 		}
/******/ 	
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
```

Do I need to do any modifications in my Aurelia-Slickgrid lib itself? Also note that I didn't update the WebPack in that lib in a long time, in fact it still uses WebPack 3 and the Aurelia-CLI is at version `1.2.3`

**Comment by 3cp** [comment]
I can see the error, probably something changed between zeit/ncc and vercel/ncc. I will check.

**Comment by 3cp** [comment]
Behold, the old zeit/ncc compiled file works.
I could not find any related issue on @vercel/ncc yet.

**Comment by 3cp** [comment]
@ghiscoding I downgraded ncc from 0.24 to 0.23, you can try again with latest code.

**Comment by ghiscoding** [comment]
@3cp wow this is awesome, you are helping a lot and I greatly appreciate it. I can confirm that it works in 2/3 of my libs (Slickgrid-Universal and Aurelia-Slickgrid) and I will confirm Angular-Slickgrid tomorrow during working hours since I only have it installed for work but I'm assuming it will pass just the same since Angular uses WebPack as well. 

Again thanks a lot for all of this, it looks quite promising 😉

**Comment by ghiscoding** [comment]
So I tried in Angular-Slickgrid and it doesn't work 😢 
It says that it cannot find `__dirname` but I don't know why, I know that in Angular they use WebPack but the config is mostly hidden and we just pass some flags in the CLI. The only thing that I can think of would be to release a `major` version of `excel-builder-webpacker` and use that only in my other 2 libs except Angular, I'll probably do that to get further... what do you guys think? Unless someone knows how to fix the `__dirname`

**Comment by 3cp** [comment]
That's unfortunate.
`__dirname` is the commonjs global which all bundlers know how to deal with it. You probably can find some related issue in angular.

**Comment by ghiscoding** [comment]
Yeah I searched a bit but couldn't find much, that's ok I think I'll opt for merging your code and bump a major version `2.x` and use it in the 2 libs that works and keep Angular with the previous `1.x` version that works for it. I was just curious if you had any other suggestions, but no big deal, I'll just do that `major` version and make everyone happy 😉

**Comment by zewa666** [comment]
I'm not sure since I don't work with Webpack. But I remember having seen something like that in Angular back in v2 or v4. Anyways it was along the lines of configuring webpack for [supporting node.__dirname](https://webpack.js.org/configuration/node/#node__dirname)

so it would look like in your webpack config:

```
{
  node: {
    __dirname: true
  }
}
```

Now I know that by default they don't expose the webpack config with their CLI but perhaps thats a guiding point?

**Comment by 3cp** [comment]
I am surprised angular toolchain didn't turn on commonjs compatibility by default.

**Comment by 3cp** [comment]
FYI, Rob released aurelia-cli v2.0.1 with that cli-bundler patch.

**Comment by ghiscoding** [comment]
Thanks I took a quick look at the Angular CLI [build commands](https://angular.io/cli/build) and couldn't find anything like that.
No worries, I'll just go straight with the `major 2.x` for 2 of my libs and stick with previous for Angular, that is an acceptable path for me, that ways it won't break Angular-Slickgrid and we'll get Aurelia-Slickgrid working with the new CLI `2.x`. 

I might be able to look at that tonight, I got another PR to finish before releasing a new version and I might be able to do all of that tonight, we'll see

**Comment by ghiscoding** [comment]
@3cp what is that cli-bundler patch you're referring to? Is that related to this discussion? Just curious

**Comment by 3cp** [comment]
The bug https://github.com/ghiscoding/aurelia-slickgrid-demos/issues/2#issuecomment-689196541

**Comment by ghiscoding** [comment]
@zewa666 
Alright I'm done with newer `excel-builder-webpacker` and Aurelia-Slickgrid newer version [2.21.0](https://github.com/ghiscoding/aurelia-slickgrid/releases) (there's a few other changes that you can see in the release, [2.21.0](https://github.com/ghiscoding/aurelia-slickgrid/releases)). The last thing I'd like you to do (if you want to help) would be to test with newer Aurelia-CLI `2.0.1` (I didn't do that yet) and perhaps create a PR in the demo, I tested and rebuilt the `Aurelia-Slickgrid-Demos` but I didn't update to the latest Aurelia-CLI (I basically always rebuild and do a prod build for both demos I have, [Aurelia-Slickgrid with BS3](https://ghiscoding.github.io/aurelia-slickgrid/#/slickgrid) and [Aurelia-Slickgrid with BS4](https://ghiscoding.github.io/aurelia-slickgrid-demos/#/slickgrid), they are both created with WebPack bundler though)

@3cp 
Thanks a lot for all of this, I was kinda lost with this `ncc`, not always fun to work with old libs, glad you helped :wink

Great to see so much collaboration 😄

**Comment by 3cp** [comment]
You are welcome. Suggest to redo your demos with latest cli generated projects.

Systemjs demo can be removed, it's removed from cli template. Because aurelia only has support on very very outdated systemjs version, aurelia doesn't work with latest systemjs (due to aurelia loader uses private systemjs api).
bs3 demos can be removed I think, it's way outdated.

You should also update your doc, there is no special config needed anymore in aurelia.json for cli-bundler.

**Comment by ghiscoding** [comment]
Thanks for suggestions, I'll wait for @zewa666 feedback on some of these things. BS3 is still use quite a lot, I'm thinking to remove it only after BS5 is out. 

> You should also update your doc, there is no special config needed anymore in aurelia.json for cli-bundler.

Do you mean the entire section in the HOWTO [For Aurelia-CLI with RequireJS/SystemJS](https://github.com/ghiscoding/aurelia-slickgrid/wiki/HOWTO--Step-by-Step#2-for-aurelia-cli-with-requirejssystemjs-modify-the-aureliajson-file)? If that is the entire thing that becomes unnecessary, then I must say you guys have made huge improvements in that regards. 😮 

Again I'll let @zewa666 confirm some of these, I had more than enough for today, long day...
Cheers

**Comment by 3cp** [comment]
> Do you mean the entire section in the HOWTO [For Aurelia-CLI with RequireJS/SystemJS](https://github.com/ghiscoding/aurelia-slickgrid/wiki/HOWTO--Step-by-Step#2-for-aurelia-cli-with-requirejssystemjs-modify-the-aureliajson-file)? If that is the entire thing that becomes unnecessary, then I must say you guys have made huge improvements in that regards. 😮

Yes, it was almost two years ago. http://aurelia.io/blog/2018/10/08/release-aurelia-cli-with-auto-tracing-bundler
For Aurelia 2, we removed support of cli-bundler, I wrote a better one: [dumber bundler](https://dumber.js.org) which also supports current Aurelia 1.

**Comment by zewa666** [comment]
Ok so here is what I did:

1. Scaffold a new project using
`npx makes aurelia/v1 slickgrid-demo -s cli-bundler,typescript,scaffold-navigation`

2.  `npm install aurelia-slickgrid`

3. add `"esModuleInterop": true,` to tsconfig.json

4. add to main.ts
```ts
import process from 'process';
window.process = process;
import {Buffer} from 'buffer';
(window as any).Buffer = Buffer;

...

configure(aurelia:Aurelia): void {
  ...
  aurelia.use.plugin('aurelia-slickgrid');
}

```

5. npm start

---

Failed with two errors:
```
Uncaught RangeError: Maximum call stack size exceeded
    at require.js:134
    at require.js:1189
    at each (require.js:59)
    at Module.emit (require.js:1188)
    at Module.<anonymous> (require.js:1149)
    at require.js:134
    at require.js:1189
    at each (require.js:59)
    at Module.emit (require.js:1188)
    at Module.<anonymous> (require.js:1149)

Uncaught (in promise) TypeError: Object prototype may only be an Object or null: undefined
    at Function.create (<anonymous>)
    at inherits (inherits_browser.js:6)
    at Object.<anonymous> (_stream_duplex.js:46)
    at Object.execCb (require.js:1696)
    at Module.check (require.js:878)
    at Module.<anonymous> (require.js:1139)
    at require.js:134
    at require.js:1189
    at each (require.js:59)
    at Module.emit (require.js:1188)
```

Here is the repo: https://github.com/zewa666/slickgrid-cli-error

Once we get this working I can gladly update any docs or samples via a PR

**Comment by ghiscoding** [comment]
Oh I thought the infinite loop was fixed, since @3cp mentioned it in this earlier [post](https://github.com/ghiscoding/aurelia-slickgrid-demos/issues/2#issuecomment-689196541)

**Comment by ghiscoding** [comment]
I did a quick test over lunch and I could replicate and I have the same errors showing up in the console.

However, I just did a quick trial of my previous [aurelia-slickgrid-demos](https://github.com/ghiscoding/aurelia-slickgrid-demos) for both RequireJS/SystemJS and after updating the `aurelia.json` file with this
```json
"dependencies": [
         {
            "name": "excel-builder-webpacker",
            "path": "../node_modules/excel-builder-webpacker",
            "main": "dist/index",
            "deps": [
              "jszip",
              "lodash"
            ]
          }
]
```
They both work with the new `excel-builder-webpacker` build using `ncc` that @3cp created
You can git clone them and try them, you'll see that they work and the Excel Export work as well.
It seems the CLI 2.x still has some kind of problem detecting some of the resources.

**Comment by zewa666** [comment]
That's cool @ghiscoding since we still can provide manual overrides with CLI 2.x I'll give your sample a try tomorrow morning and let you know

**Comment by 3cp** [comment]
Only cli v2.0.1 fixed the Maximum call stack issue. You don't need manual config for excel-builder-webpacker.

**Comment by ghiscoding** [comment]
@3cp @zewa666 
I added all dependencies manually and found `jszip` as being the culprit, or at least adding it manually makes the build to work in @zewa666 fork, I add it to the `dependencies` like this
```json
{
  "name": "jszip",
   "path": "../node_modules/jszip/dist",
   "main": "jszip.min"
}
```
@3cp since `jszip` is **only** required by the `excel-builder`, are there any other things that can be done with the `ncc` build? Or do we just add that 1 package into the `dependencies` array and be done with it?

**Comment by 3cp** [comment]
@ghiscoding yes, it's easy. You need to remove `-e jszip` from the build command, and move `jszip` from excel-builder package.json dependencies to devDependencies.
You probably can do same thing for `q`.

But don't remove `-e lodash`, lodash npm package is in UMD format, it will cause side effect inside ncc bundle.

**Comment by ghiscoding** [comment]
Hmm I tried that and I get the same result or actually worst since manually adding `jszip` to `aurelia.json` no longer works, so I basically still get the same error with/without `jszip` in the `aurelia.json`

`Uncaught RangeError: Maximum call stack size exceeded`

**Comment by 3cp** [comment]
1. for the manual config, jszip package.json has this
```
"browser": {
    "readable-stream": "./lib/readable-stream-browser.js",
    ".": "./dist/jszip.min.js"
  },
```

I think cli-bundler didn't handle `"."` correctly, I will read the spec again. Ideally the manually config is not needed.

2. for the ncc bundle including jszip. I am not sure why it still has the problem, I will spend sometime to dig into that.

**Comment by 3cp** [comment]
I fixed the cli bug, will be in next patch release.

**Comment by 3cp** [comment]
@zewa666 I confirm the coming cli bug can fix the final Maximum call stack size exceeded.

**Comment by 3cp** [comment]
@ghiscoding you can also bypass this issue now by modifying `excel-builder.js/src/excel-builder.js`.

from
```js
var JSZip = require('jszip');
```
to
```js
var JSZip = require('jszip/dist/jszip.min.js');
```

I don't understand why there is problem with ncc bundled with jszip source code, but this fix forces it to use `jszip/dist/jszip.min.js` file (for both situations: external jszip, or bundled jszip).

**Comment by zewa666** [comment]
So let me please say first thanks to both of you. This is fantastic and with @ghiscoding's approach of manually adding jszip in the dependencies it definitely works perfectly.

I'm going to wait for @3cp's latest release of the CLI and try it out then. If that works we'll be able to manage it solely with that process and buffer workaround and that is perfectly fine in my opinion. Once that is checked I'll create a PR for the current CLI sample and update it to latest version plus a change for the howto step by step wiki page.

**Comment by zewa666** [comment]
btw. @ghiscoding could you point me to the right repo where I should create these type of issue-requests for slickgrid? I wasnt sure when searching and landed here initially but e.g now I'd have a minor thing to ask about a SCSS compile issue, so where would you feel this should go?

**Comment by ghiscoding** [comment]
@zewa666 if it's anything with [aurelia-slickgrid](https://github.com/ghiscoding/aurelia-slickgrid), which it is for scss, then plz open an issue there.

**Comment by ghiscoding** [comment]
@zewa666 the fix that @3cp pushed couple days ago got released in Aurelia-CLI today as can be seen in the [aurelia-cli - releases](https://github.com/aurelia/cli/releases) and I tried it and it works on my end. I guess we could close this issue afterward. 

Thanks a ton to both of you for fixing this.

**Comment by zewa666** [comment]
Cool, will test on Monday but fine to close for now. I'll reply here if any issues exist and follow up with the samples PR

**Comment by 3cp** [comment]
Cheers!

**Comment by zewa666** [comment]
Sorry, totally forgot to confirm that now everything works. There are still some parts needed for bootstrap v3 and in both examples for jquery-ui-dist but see about that in the PR I've opened as those are expected to happen.
